autoprecompiles/src/memory_optimizer.rs

```diff
@@ -63,6 +93,12 @@ fn redundant_memory_interactions_indices<T: FieldElement>(
                 continue;
```

Comment:
I think we only need to clear the entries for memory type for which that happens, right?

Reply:
No, but you could add a comment, so when it does we'd know where to look.